android/app/src/main/cpp/peer_audio_manager.cpp

```
@@ -1,287 +1,465 @@
+            env->DeleteLocalRef(jData);
         }
         env->DeleteLocalRef(callbackClass);
     }
-
-    if (needDetach) {
-        jvm->DetachCurrentThread();
-    }
+    env->DeleteLocalRef(callback);
 }
 
 void PeerAudioManager::setCallback(JNIEnv* env, jobject callback) {
```

Clang (14.0.6)

[warning] method 'setCallback' can be made static

(readability-convert-member-functions-to-static)